runtime/shibuya/src/xcm_config.rs

```diff
@@ -260,3 +271,72 @@ impl cumulus_pallet_dmp_queue::Config for Runtime {
     type XcmExecutor = XcmExecutor<XcmConfig>;
     type ExecuteOverweightOrigin = EnsureRoot<AccountId>;
 }
+
+/// Convert `AccountId` to `MultiLocation`.
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+    fn convert(account: AccountId) -> MultiLocation {
+        X1(AccountId32 {
+            network: None,
+            id: account.into(),
+        })
+        .into()
+    }
+}
+
+parameter_types! {
+    /// The absolute location in perspective of the whole network.
+    pub ShibuyaLocationAbsolute: MultiLocation = MultiLocation {
+        parents: 1,
+        interior: X1(
+            Parachain(ParachainInfo::parachain_id().into())
+        )
+    };
+    /// Max asset types for one cross-chain transfer. `2` covers all current use cases.
+    /// Can be updated with extra test cases in the future if needed.
+    pub const MaxAssetsForTransfer: usize = 2;
+}
+
+/// Convert `AssetId` to optional `MultiLocation`. The impl is a wrapper
+/// on `ShibuyaAssetLocationIdConverter`.
+pub struct AssetIdConvert;
+impl Convert<AssetId, Option<MultiLocation>> for AssetIdConvert {
+    fn convert(asset_id: AssetId) -> Option<MultiLocation> {
+        ShibuyaAssetLocationIdConverter::reverse_ref(&asset_id).ok()
+    }
+}
+
+/// `MultiAsset` reserve location provider. It's based on `RelativeReserveProvider` and in
+/// addition will convert self absolute location to relative location.
+pub struct AbsoluteAndRelativeReserveProvider<AbsoluteLocation>(PhantomData<AbsoluteLocation>);
+impl<AbsoluteLocation: Get<MultiLocation>> Reserve
+    for AbsoluteAndRelativeReserveProvider<AbsoluteLocation>
+{
+    fn reserve(asset: &MultiAsset) -> Option<MultiLocation> {
+        RelativeReserveProvider::reserve(asset).map(|reserve_location| {
```

Comment:
Had to check what this does, but it seems it's functionally (almost?) same as: https://github.com/AstarNetwork/astar-frame/blob/polkadot-v0.9.39/primitives/xcm/src/lib.rs#L205

Not for this PR, but let's get rid of our _custom_ implementation, if we can use the ORML one which does the exact same thing.

Reply:
I agree! They are pretty similar. The `ReserveAssetFilter` in XCM primitives can be refactored to be leveraged on [`AbsoluteReserveProvider`](https://github.com/open-web3-stack/open-runtime-module-library/blob/a4977fe70eeea93b072706f7ab2d2577f4a8d320/traits/src/location.rs#L49) impl in ORML.

Btw I'm aware that some of the config could be moved to `xcm-primitives` and shared in different runtimes. I'm thinking of doing it after Shibuya integration finished and tested, so there would be less back and forth with `astar-frame` PRs, in case anything didn't work as expected. Hope it makes sense.

Reply:
Of course! :slightly_smiling_face: 

I also saw your comment about merging `Astar` and `astar-frame`, so that's something we can discuss soon too!